File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Classes/FontAwesomeIcon.java
package com.alphadevelopmentsolutions.frcscout.Classes;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.util.AttributeSet;
import android.view.View;

public abstract class FontAwesomeIcon extends android.support.v7.widget.AppCompatTextView
{
    public FontAwesomeIcon(Context context)
    {
        super(context);
    }

    public FontAwesomeIcon(Context context, AttributeSet attrs)
    {
        super(context, attrs);
    }

    public FontAwesomeIcon(Context context, AttributeSet attrs, int defStyleAttr)
    {
        super(context, attrs, defStyleAttr);
    }

    /**
     * Hides the fontawesome icon from the screen
     */
    public void hide()
    {
        setVisibility(GONE);
    }

    /**
     * Sets the URL to go to when the icon has been clicked
     */
    public void setURL(final String URL, final Context context)
    {
        setOnClickListener(new OnClickListener()
        {
            @Override
            public void onClick(View v)
            {
                Intent intent = new Intent(Intent.ACTION_VIEW);
                intent.setData(Uri.parse(URL));
                context.startActivity(intent);
            }
        });
    }

    abstract void init();
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Classes/FontAwesomeIcon.kt
package com.alphadevelopmentsolutions.frcscout.Classes

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.util.AttributeSet
import android.view.View

abstract class FontAwesomeIcon : android.support.v7.widget.AppCompatTextView
{
    constructor(context: Context) : super(context)
    {
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    {
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
    {
    }

    /**
     * Hides the fontawesome icon from the screen
     */
    fun hide()
    {
        visibility = View.GONE
    }

    /**
     * Sets the URL to go to when the icon has been clicked
     */
    fun setURL(URL: String, context: Context)
    {
        setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW)
            intent.data = Uri.parse(URL)
            context.startActivity(intent)
        }
    }

    internal abstract fun init()
}
